src/main/java/duke/maincomponents/Parser.java

```diff
@@ -0,0 +1,197 @@
+package duke.maincomponents;
+
+import java.util.ArrayList;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ShowListCommand;
+import duke.command.ToDoCommand;
+import duke.exceptions.DukeException;
+
+
+public class Parser {
+
+    private static boolean checkStringStartingEquals(String userInput, String stringToCheck) {
+        return userInput.length() >= stringToCheck.length()
+                && userInput.substring(0, stringToCheck.length()).equals(stringToCheck);
+    }
+
+    private static boolean checkStringEquals(String userInput, String stringToCheck) {
+        return userInput.equals(stringToCheck) || userInput.equals(stringToCheck + " ");
+    }
+
+    private boolean equalsToList(String userInput) {
+        return checkStringEquals(userInput, "list");
+    }
+
+    private boolean equalsToDone(String userInput) {
+        return checkStringStartingEquals(userInput, "done");
+    }
+
+    public int parseDoneCommand(String userInput) throws DukeException {
```

Comment:
Perhaps you could provide a descriptive header for this method, as well as other public methods?